include/circt/Dialect/HW/HWTypeInterfaces.td

```diff
@@ -29,14 +29,12 @@ def FieldIDTypeInterface : TypeInterface<"FieldIDTypeInterface"> {
       Get the sub-type of a type for a field ID, and the subfield's ID. Strip
       off a single layer of this type and return the sub-type and a field ID
       targeting the same field, but rebased on the sub-type.
-    }], "std::pair<circt::hw::FieldIDTypeInterface, uint64_t>",
-    "getSubTypeByFieldID", (ins "uint64_t":$fieldID)>,
 
-    InterfaceMethod<[{
-      Return the final type targeted by this field ID by recursively walking all
-      nested aggregate types. This is the identity function for ground types.
-    }], "circt::hw::FieldIDTypeInterface", "getFinalTypeByFieldID",
-    (ins "uint64_t":$fieldID)>,
+      The resultant type may not be a FieldIDTypeInterface if the resulting 
+      fieldID is zero.  This means that leaf types may be ground without 
```

Comment:
Hmm, having 0 fields is not quite same as ground (0-element vector, empty bundle), at least in the FIRRTL sense.
No suggested change here, but wanted to note that since it's a small detail easily missed when using FieldIDTypeInterface.